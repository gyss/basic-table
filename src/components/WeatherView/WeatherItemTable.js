/** @jsx jsx */

import React from 'react'
import {css, jsx} from '@emotion/core'
import PropTypes from 'prop-types'
import {AutoSizer, Table, Column} from 'react-virtualized'

function rowClassName({index}) {
  if (index < 0) {
    return 'headerRow'
  } else {
    return index % 2 === 0 ? 'evenRow' : 'oddRow'
  }
}

const WeatherItemTable = ({item}) => {
  return (
    <AutoSizer>
      {({height, width}) => (
        <Table
          width={Math.max(550, width)}
          height={height}
          headerHeight={40}
          rowHeight={30}
          rowCount={item.weather.length}
          rowGetter={({index}) => item.weather[index]}
          rowClassName={rowClassName}>
          <Column label="Main" dataKey="main" width={150} />
          <Column width={200} label="Description" dataKey="description" flexGrow={1} />
          <Column width={200} label="Icon" dataKey="icon" />
        </Table>
      )}
    </AutoSizer>
  )
}

WeatherItemTable.propTypes = {
  item: PropTypes.shape({
    weather: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        main: PropTypes.string,
        description: PropTypes.string,
        icon: PropTypes.string,
      }),
    ),
  }).isRequired,
}

export default WeatherItemTable
