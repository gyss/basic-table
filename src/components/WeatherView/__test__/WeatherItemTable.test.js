/** @format */

import * as React from 'react'
import {render, screen, fireEvent} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import WeatherItemTable from '../WeatherItemTable'
import weatherFixture from '../../../fixtures/weatherFixture'

describe('<WeatherItemTable>', () => {
  let props

  beforeEach(() => {
    props = {
      onClick: jest.fn(),
    }
  })

  test('should render', () => {
    const {asFragment} = render(<WeatherItemTable item={weatherFixture.list[0]} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
