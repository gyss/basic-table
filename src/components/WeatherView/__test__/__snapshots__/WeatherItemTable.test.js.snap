// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherItemTable> should render 1`] = `
<DocumentFragment>
  <div
    style="overflow: visible; height: 0px; width: 0px;"
  >
    <div
      aria-colcount="3"
      aria-rowcount="1"
      class="ReactVirtualized__Table"
      role="grid"
    >
      <div
        class="ReactVirtualized__Table__headerRow headerRow"
        role="row"
        style="height: 40px; overflow: hidden; padding-right: 0px; width: 550px;"
      >
        <div
          class="ReactVirtualized__Table__headerColumn"
          role="columnheader"
          style="flex: 0 1 150px;"
        >
          <span
            class="ReactVirtualized__Table__headerTruncatedText"
            title="Main"
          >
            Main
          </span>
        </div>
        <div
          class="ReactVirtualized__Table__headerColumn"
          role="columnheader"
          style="flex: 1 1 200px;"
        >
          <span
            class="ReactVirtualized__Table__headerTruncatedText"
            title="Description"
          >
            Description
          </span>
        </div>
        <div
          class="ReactVirtualized__Table__headerColumn"
          role="columnheader"
          style="flex: 0 1 200px;"
        >
          <span
            class="ReactVirtualized__Table__headerTruncatedText"
            title="Icon"
          >
            Icon
          </span>
        </div>
      </div>
      <div
        aria-label="grid"
        class="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
        role="rowgroup"
        style="box-sizing: border-box; direction: ltr; height: -40px; position: relative; width: 550px; will-change: transform; overflow-x: hidden; overflow-y: auto;"
        tabindex="0"
      />
    </div>
  </div>
  <div
    class="resize-triggers"
  >
    <div
      class="expand-trigger"
    >
      <div
        style="width: 1px; height: 1px;"
      />
    </div>
    <div
      class="contract-trigger"
    />
  </div>
</DocumentFragment>
`;
